services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/discussion/DiscussionUtilsTest.java

```diff
@@ -46,17 +46,11 @@ public void testGetMentionedUsernameCaseWhiteSpacePostfix(){
 	@Test
 	public void testGetMentionedUsernameCaseDoubleAlpha(){
 		Set<String> expected = new HashSet<String>();
-		expected.add("anonymous");
+		expected.add("anon");
+		expected.add("ymous");
 		assertEquals(expected, DiscussionUtils.getMentionedUsername("@anon@ymous"));
```

Comment:
The fix in this PR breaks this test, so I changed the expected result.

I don't quite understand why this test existed in it's previous state. Is there a reason to preserve the old behavior?

Reply:
I think it works better now.  Good job!

Reply:
I do have much time to look at this now, but please do check how we use the parsed username to look up users. We may have done this because we are using services that list prefix of username. So if the username is not exact, we may get both user "kim" and "kimyen" for "@kim".

Reply:
Thanks @nickgros for taking a look! I appreciate it. I should have looked at the changes before making any comments.